hyperspy/misc/array_tools.py

```diff
@@ -470,8 +470,8 @@ def numba_closest_index_round(axis_array,value_array):
     index_array = np.empty_like(value_array,dtype='uint')
     #assign on flat, iterate on flat.
     for i,v in enumerate(value_array.flat):
-        index_array.flat[i] = np.abs(axis_array - v).argmin()
-
+        vdiff_array = np.abs(axis_array - v)
+        index_array.flat[i] = np.flatnonzero(vdiff_array == np.min(vdiff_array))[-1]
```

Comment:
Here the idea is to deal with the case where two identical values are found in `vdiff_array`  in which case you want to return the last one to ensure rounding above, I guess.

Unfortunately, while this might solve a failing test, this is not a solution for two reasons:
1. if the axis is sorted in decreasing order, this will still round below
2. the case where  `vdiff_array==np.min(vdiff_array)` matches more than 1 value does not typically happens. 

Here is an example:
```python
>>> import numpy as np
>>> ax = np.linspace(-10.0,-11.0,11)
>>> ax
array([-10. , -10.1, -10.2, -10.3, -10.4, -10.5, -10.6, -10.7, -10.8,
       -10.9, -11. ])
>>> vdiff_array = np.abs(ax+10.15)
>>> vdiff_array
array([0.15, 0.05, 0.05, 0.15, 0.25, 0.35, 0.45, 0.55, 0.65, 0.75, 0.85])
>>> vdiff_array==np.min(vdiff_array)
array([False, False,  True, False, False, False, False, False, False,
       False, False])
```
As you can see, this performs a rounding below (case 1) and we only have one single value matched in the equality comparison operator (case 2). Of course, the reason for this behaviour is float arithmetics. We have:

```python
>>> vdiff_array[1] 
0.05000000000000071
>>> vdiff_array[2]
0.049999999999998934
```

which, no matter how you look at it, will still round below in most strategies, I think the best way to deal with it would be to bias very slightly vdiff_array towards upper-rounding. I would advocate something like `np.abs(axis_array - v + MACHINEEPSILON)` and `MACHINEEPSILON` would be determined using some relative tolerance, eg `rtol=1e-10; MACHINEEPSILON = np.min(np.diff(axis))*rtol `.

I don't pretend this is the ideal solution. But it can shifts a commonly encountered problem (incorrect value2indexing exactly half the interval between axis points) to a much less commonly encountered one (incorrect value2indexing a value that is very close but not equal to exactly half the interval between axis points)

Reply:
Yes, the idea was to enforce rounding up for a `0.5` value.

I like your idea with the upwards bias, I had thought in a similar direction. I'll implement that to see.

Though, I now realize that numpy also does not necessarily round 0.5 upwards, but it rounds to the next even number: https://github.com/numpy/numpy/blob/e94ed84010c60961f82860d146681d3fd607de4e/numpy/core/fromnumeric.py#L2723-L2789 
(explicitly mentioning the issue with machine precision as well).

Reply:
Yeah exactly. 

Actually I wonder why this rounding approach is used in hyperspy. When slicing with a float value, I would personally prefer knowing in advance whether the sliced axis will contain or not the value I input.

I would agree it's not a big deal, but in the end we spend some effort to preserve this functionality.

Reply:
It is python rounding which uses the round-to-even strategy to avoid accumulation of offset: https://realpython.com/python-rounding/#pythons-built-in-round-function

Reply:
To be consistent with the behaviour for non-uniform axes, I have used a machineepsilon here as well to have 'round-away-from-zero' instead of 'round-to-even'

Reply:
Well, in the end I think it does not make sense to touch this function,because round is of course used elsewhere in the code as well and we should be consistent there.

Reply:
> To be consistent with the behaviour for non-uniform axes, I have used a machineepsilon here as well to have 'round-away-from-zero' instead of 'round-to-even'

Not sure what behaviour you mean. You want a behaviour similar to numpy.rint? https://numpy.org/doc/stable/reference/generated/numpy.rint.html#numpy.rint 

In any case, I quickly checked out your branche and I think there still is an issue of uniformity between the Uniform and Non Uniform Axes

```python
>>>from hyperspy.axes import UniformDataAxis

>>>inax = [[-11.0,-10.9],[-10.9,-11.0],[+10.9,+11.0],[+11.0,+10.9]]
>>>inval = [-10.95,-10.95,10.95,10.95]

>>>for i,j in zip(inax,inval):
>>>    ax = UniformDataAxis(scale=i[1]-i[0],offset=i[0],size=len(i))
>>>    nua_idx = super(type(ax),ax).value2index(j,rounding=round)
>>>    unif_idx = ax.value2index(j,rounding=round)
>>>    print("Value2Index IN ax: {}, val: {} --> UAX: {}, NUA: {}".format(ax.axis,j,unif_idx,nua_idx))

Value2Index IN ax: [-11.  -10.9], val: -10.95 --> UAX: 1, NUA: 0
Value2Index IN ax: [-10.9 -11. ], val: -10.95 --> UAX: 0, NUA: 1
Value2Index IN ax: [10.9 11. ], val: 10.95 --> UAX: 0, NUA: 1
Value2Index IN ax: [11.  10.9], val: 10.95 --> UAX: 1, NUA: 0

```

I assume the behaviour you desire is the one from NUA?